test/example.js

```diff
@@ -116,7 +116,15 @@ const asyncLambda = async () => {};
 
 // Async functions are supported
 // Returns: <Promise>
-const asyncFunc = async function() {};
+// Resolves: <Object>
```

Comment:
I'd prefer to avoid adding keywords for such cases. I think it'll be better to allow to customize types:
```js
// Returns: <Promise<{ a, b }>>
//   a <number>
//   b <string>
```
This would also allow to specify `Map`, `Set` types etc. (i.e. `Map<string, number>`, `Set<string>`).